jwst/pixel_replace/pixel_replace.py

```diff
@@ -0,0 +1,382 @@
+                    log.info(f"Input MRS frame had {n_replaced} total pixels replaced.")
+                else:
+                    # NRS_IFU method - Fixed number of IFU slices to iterate over
+                    for i in range(30):
+                        slice_wcs = nrs_wcs_set_input(self.input, i)
+                        _, _, wave = slice_wcs.transform('detector', 'slicer', yy, xx)
+
+                        # Define a mask that is True where this trace is located
+                        trace_mask = (wave > 0)
+                        trace_model = self.input.copy()
+                        trace_model.dq = np.where(
+                            # When not in this trace, set NON_SCIENCE and DO_NOT_USE
+                            ~trace_mask,
+                            trace_model.dq | self.DO_NOT_USE | self.NON_SCIENCE,
+                            trace_model.dq
+                        )
+
+                        trace_model = self.algorithm(trace_model)
+                        self.output.data = np.where(
+                            # Where trace is located, set replaced values
+                            trace_mask,
+                            trace_model.data,
+                            self.output.data
+                        )
+
+                        self.output.dq = np.where(
+                            # Where trace is located, set replaced values
+                            trace_mask,
+                            trace_model.dq,
+                            self.output.dq
+                        )
+
+                        n_replaced = np.count_nonzero(trace_model.dq & self.FLUX_ESTIMATED)
+                        log.info(f"Input NRS_IFU frame had {n_replaced} pixels replaced in IFU slice {i + 1}.")
+
+                        trace_model.close()
+
+                    n_replaced = np.count_nonzero(self.output.dq & self.FLUX_ESTIMATED)
+                    log.info(f"Input NRS_IFU frame had {n_replaced} total pixels replaced.")
+
+        # MultiSlitModel inputs (WFSS, NRS_FIXEDSLIT, ?)
+        elif isinstance(self.input, datamodels.MultiSlitModel):
+
+            for i, slit in enumerate(self.input.slits):
+                slit_model = datamodels.SlitModel(self.input.slits[i].instance)
+                slit_replaced = self.algorithm(slit_model)
+
+                n_replaced = np.count_nonzero(slit_replaced.dq & self.FLUX_ESTIMATED)
+                log.info(f"Slit {i} had {n_replaced} pixels replaced.")
+
+                self.output.slits[i] = slit_replaced
+
+        # CubeModel inputs are TSO (so far?); only SlitModel seen so far is NRS_BRIGHTOBJ, also requiring
+        # a re-packaging of the data into 2D inputs for the algorithm.
+        elif isinstance(self.input, (datamodels.CubeModel, datamodels.SlitModel)):
+            if len(self.input.data) != len(self.input.dq):
+                log.critical("Data and DQ arrays are not of equal length - skipping pixel replacement.")
```

Comment:
have you actually encountered this condition in some datasets?!

Reply:
I can't imagine I thought this one up, but I have no recollection of when this occurred. I can remove it and we can wait for any culprits to pop up, if they exist.